datenguide_python/query_builder.py

```diff
@@ -153,10 +203,44 @@ def get_graphql_query(self) -> str:
         else:
             raise TypeError("region or parent must be defined.")
 
-    def get_fields(self) -> List[Union[str, Field]]:
+    def get_fields(self) -> List[str]:
         """Get all fields of a query.
 
         Returns:
             List[Union[str, Field]] -- a list of strings and / or Fields
         """
-        return self.fields
+        field_list = []
+        for field in self.fields:
+            field_list.extend(Query._get_fields_helper(field))
+        return field_list
+
+    def results(self) -> Optional[ExecutionResults]:
+        return QueryExecutioner().run_query(self)
+
+    @staticmethod
+    def _get_fields_helper(field: Union[str, object]) -> List[str]:
```

Comment:
Typically one uses "Any" (imported from typing) instead of "object" for types. In this case however you could be more specific and use and use the class Field itself. You'll need to use the cast function from typing in the if statement as well to tell mypy that inside the different paths the is no type ambiguity anymore. In particular this means to use cast(str,field) or cast(Field,field) inside the branches.

Reply:
I did use "Field", but this only worked for python 3.7 not for 3.6., because I am referencing the same clas within the class description (see this: https://stackoverflow.com/questions/33533148/how-do-i-specify-that-the-return-type-of-a-method-is-the-same-as-the-class-itsel)
when I used the from __future__ import annotations Travis would through me an error for python 3.6
Do you know a better way to handle this?

Reply:
Future should not be necessary as that refers to python 4 if I'm not mistaken. I was thinking about something along the following lines, which works for me in a 3.6 env, i.e. mypy does not yield any errors. Let me know if this does not fit your problem, because I also do not fully understand what you mean with referencing the class within the *class description*, I'm only talking about the class definition and am currently not aware of issues with the description.

```python
from typing import Union, Any, cast

class A:
    def __init__(self,a:int) ->None:
        self.a = a 

    def foo(self,a: Union[int,"A"]) -> int:
            if type(a) == int:
                return cast(int,a)
            else:
                return cast("A",a).a + self.a

b = A(5)
print(b.foo(1))
print(b.foo(b))
```